tests/integration/security/jwt_test.go

```diff
@@ -66,251 +63,290 @@ func TestRequestAuthentication(t *testing.T) {
 			ctx.Config().ApplyYAMLOrFail(t, ns.Name(), jwtPolicies...)
 			defer ctx.Config().DeleteYAMLOrFail(t, ns.Name(), jwtPolicies...)
 
-			var a, b, c, d, e, f echo.Instance
-			echoboot.NewBuilder(ctx).
-				With(&a, util.EchoConfig("a", ns, false, nil, nil)).
-				With(&b, util.EchoConfig("b", ns, false, nil, nil)).
-				With(&c, util.EchoConfig("c", ns, false, nil, nil)).
-				With(&d, util.EchoConfig("d", ns, false, nil, nil)).
-				With(&e, util.EchoConfig("e", ns, false, nil, nil)).
-				With(&f, util.EchoConfig("f", ns, false, nil, nil)).
-				BuildOrFail(t)
+			aSet := apps.A.Match(echo.Namespace(ns.Name()))
+			bSet := apps.B.Match(echo.Namespace(ns.Name()))
+			cSet := apps.C.Match(echo.Namespace(ns.Name()))
+			dSet := apps.D.Match(echo.Namespace(ns.Name()))
+			eSet := apps.E.Match(echo.Namespace(ns.Name()))
+			fSet := apps.F.Match(echo.Namespace(ns.Name()))
 
-			testCases := []authn.TestCase{
-				{
-					Name: "valid-token-noauthz",
-					Request: connection.Checker{
-						From: a,
-						Options: echo.CallOptions{
-							Target:   c,
-							PortName: "http",
-							Scheme:   scheme.HTTP,
-							Headers: map[string][]string{
-								authHeaderKey: {"Bearer " + jwt.TokenIssuer1},
+			callCount := 1
+			if ctx.Clusters().IsMulticluster() {
+				// so we can validate all clusters are hit
+				callCount = util.CallsPerCluster * len(ctx.Clusters())
+			}
+
+			for _, cluster := range ctx.Clusters() {
+				ctx.NewSubTest(fmt.Sprintf("From a and b In %s", cluster.Name())).Run(func(ctx framework.TestContext) {
```

Comment:
nit: I think just `"from %s", clusterName` might be more clear. Maybe group the from "a tests" and the "from b" in another level of subtest or include the source service in the test name.

Reply:
Since there is no iteration between src services in test cases, I will skip the further grouping in this PR.